### PR TITLE
Remove Typescript and Upgrade Dependencies

### PR DESCRIPTION
**Pull request by nublikasa**

### CODE REVIEW

File: src/components/Alert/Alert.js [hunk 1]
```diff
@@ -2,14 +2,36 @@
...
 */
-import * as classNames from 'classnames';
-import * as React from 'react';
+import classNames from 'classnames';
+import React from 'react';
 
 import * as Rivet from '../util/Rivet';
 import Icon from '../util/RivetIcons';
-import { StatelessAlertProps, statelessPropTypes } from './common';
+import PropTypes from 'prop-types';
```
Comment (athill) [inline src/components/Alert/Alert.js]: I'm not saying all our imports will be perfect with so many changes and don't need to update here, but as a rule, we usually have external dependencies (in alpha order), followed by a blank line, followed by local dependencies (possibly followed by a blank line then CSS imports)

File: src/components/Checkbox/Checkbox.js [hunk 2]
```diff
@@ -2,21 +2,24 @@
...
+import PropTypes from 'prop-types';
 import * as Rivet from '../util/Rivet';
 
-export interface CheckboxProps {
+const propTypes = {
     /**
      * The checkbox label
      */
-    label: string,
+    label: PropTypes.string.isRequired,
```
Comment (athill) [inline src/components/Checkbox/Checkbox.js]: We can address this in another pass, but I think all labels should be `PropTypes.node`. Someone inevitably is going to want to use markup in their labels.

File: src/components/Pagination/Pagination.js [hunk 3]
```diff
@@ -2,37 +2,38 @@
...
-interface PaginationItemProps {
-  ['aria-current']: string;
-  ['aria-disabled']: boolean;
-  disabled: boolean;
-}
+// interface PaginationItemProps {
+//   ['aria-current']: string;
+//   ['aria-disabled']: boolean;
+//   disabled: boolean;
+// }
```
Comment (athill) [inline src/components/Pagination/Pagination.js]: Please remove this commented out code

File: src/components/Tabs/Tabs.js [hunk 4]
```diff
@@ -47,25 +47,29 @@ class Tabs extends React.PureComponent<
...
-
-    !tabs[prev] ? tabs[tabs.length - 1].focus() : tabs[prev].focus();
+  focusPreviousTab(tabs, target) {
+    let prevIndex = tabs.indexOf(target) - 1;
+    if (prevIndex < 0) {
+      prevIndex = tabs.length - 1;
+    }
+    // console.log(tabs[prevIndex]);
+    tabs[prevIndex].focus();
+    // this.changeTab(prevIndex);
```
Comment (athill) [inline src/components/Tabs/Tabs.js]: Please remove these commented out lines

File: src/components/Tabs/Tabs.test.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    // // Right arrow key should focus first tab if last tab has focus
+    // it('Should focus the first tab when last tab has focus and the right key is pressed', () => {
+    //   simulateKeyboardInteraction({
+    //     component: cut,
+    //     start: 'button#t-three-tab',
+    //     end: 'button#t-one-tab',
+    //     key: 39
+    //   });
+    // });
+
```
Comment (athill) [inline src/components/Tabs/Tabs.test.js]: Oooh, these should be uncommented and fixed if necessary

File: src/components/util/Rivet.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+//     //
+//     xxs?: Edge | Edge[],
+//     xs?:  Edge | Edge[],
+//     sm?: Edge | Edge[],
+//     md?: Edge | Edge[],
+//     lg?: Edge | Edge[],
+//     xl?: Edge | Edge[],
+//     xxl?: Edge | Edge[],
+// }
+
```
Comment (athill) [inline src/components/util/Rivet.js]: Please remove commented out code

File: src/components/Alert/Alert.js (hunk 1 above)
Comment (nublikasa) [reply src/components/Alert/Alert.js]: 👍 Move Prop-Types import to be higher

File: src/components/Checkbox/Checkbox.js (hunk 2 above)
Comment (nublikasa) [reply src/components/Checkbox/Checkbox.js]: Changed to node

File: src/components/Grid/Col.js [hunk 7]
```diff
@@ -2,118 +2,131 @@
...
 */
-import * as classNames from 'classnames';
-import * as React from 'react';
+import classNames from 'classnames';
+import React from 'react';
 import * as Rivet from '../util/Rivet';
+import PropTypes from 'prop-types';
 
-interface ColProps {
+const numList = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
```
[the remark this answers is not in the document]
Comment (nublikasa) [reply src/components/Grid/Col.js]: I have renamed the variable. Good suggestion :)

File: src/components/Pagination/Pagination.js (hunk 3 above)
Comment (nublikasa) [reply src/components/Pagination/Pagination.js]: Good catch. Done.

File: src/components/Tabs/Tabs.js (hunk 4 above)
Comment (nublikasa) [reply src/components/Tabs/Tabs.js]: Done

File: src/components/Pagination/Pagination.js [hunk 8]
```diff
@@ -2,37 +2,38 @@
...
-  size?: 'small';
+  size: PropTypes.oneOf(['small'])
 }
 
-interface PaginationItemProps {
-  ['aria-current']: string;
-  ['aria-disabled']: boolean;
-  disabled: boolean;
-}
+// interface PaginationItemProps {
```
Comment (burnumd) [inline src/components/Pagination/Pagination.js]: If this isn't used, it should be removed.

File: src/setupTests.js [hunk 9]
```diff
@@ -10,6 +10,6 @@ window.requestAnimationFrame = function (callback) {
     return 0;
   };
 
-import * as Enzyme from 'enzyme';
-import * as Adapter from 'enzyme-adapter-react-16';
+import Enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
 Enzyme.configure({ adapter: new Adapter() });
```
Comment (burnumd) [inline src/setupTests.js]: This file needs a newline at the end.

File: src/components/util/Rivet.js (hunk 6 above)
Comment (wsams) [reply src/components/util/Rivet.js]: This code has been removed and validation moved to the appropriate functions.

File: src/components/Tabs/Tabs.test.js (hunk 5 above)
Comment (wsams) [reply src/components/Tabs/Tabs.test.js]: These are now uncommented and all pass.

File: public/index.html [hunk 10]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (wsams) [reply public/index.html]: We just resolve the build. Looks like `public/index.html` just needs to exist in order for `react-scripts build` to succeed. Couldn't find a way to not require this file.

**Comment by jhoerr** [comment]
@nublikasa Can you say a little bit about what you mean by "remove typescript"?

**Comment by nublikasa** [comment]
This PR is no longer needed.

**Pull request closed without merging**